### PR TITLE
Enable item option acquisition in item summaries

### PR DESCRIPTION
**Pull request by CarlosFdez**
Attempts to tweak it to a get document and a get actor phase, making it more predictable and also work better with items as parameters.

### CODE REVIEW

**Comment by CarlosFdez** [comment]
I kinda realized as I was putting this up that maybe now that Kineticist is getting released and we don't have time for actions that act, maybe we outta consider opening the item retrieval to more things. Was there something we *didn't* want it to grab? I believe it was just weapon roll notes.

File: src/scripts/ui/inline-roll-links.ts [hunk 1]
```diff
@@ -256,13 +257,9 @@ export const InlineRollLinks = {
...
 
-        const speaker =
-            foundryDoc instanceof ActorPF2e
-                ? ChatMessagePF2e.getSpeaker({
-                      actor: foundryDoc,
-                      token: foundryDoc.token ?? foundryDoc.getActiveTokens(false, true).shift(),
-                  })
-                : ChatMessagePF2e.getSpeaker();
+        const speaker = actor
+            ? ChatMessagePF2e.getSpeaker({ actor, token: actor.token ?? actor.getActiveTokens(false, true).shift() })
```
Comment (stwlam) [inline src/scripts/ui/inline-roll-links.ts:261]:
```suggestion
            ? ChatMessagePF2e.getSpeaker({ actor, token: actor.getActiveTokens(false, true).shift() })
```

File: src/scripts/ui/inline-roll-links.ts [hunk 2]
```diff
@@ -289,23 +286,22 @@ export const InlineRollLinks = {
...
-        const sheet: { id?: string; document?: unknown; actor?: unknown; journalEntry?: unknown } | null =
-            ui.windows[Number(html.closest<HTMLElement>(".app.sheet")?.dataset.appid)];
+    const sheet: { id?: string; document?: unknown } | null =
+        ui.windows[Number(html.closest<HTMLElement>(".app.sheet")?.dataset.appid)] ?? null;
 
-        return sheet.document instanceof ActorPF2e || sheet.document instanceof JournalEntry ? sheet.document : null;
-    },
-};
+    const document = sheet?.document;
+    return document && (document instanceof ActorPF2e || document instanceof JournalEntry) ? document : null;
```
Comment (stwlam) [inline src/scripts/ui/inline-roll-links.ts:299]:
```suggestion
    return document instanceof ActorPF2e || document instanceof JournalEntry ? document : null;
```

Comment (CarlosFdez) [reply src/scripts/ui/inline-roll-links.ts:299]: I should have scrutinized the original logic more closely than I did whoopsie

**Pull request merged**